crates/ra_project_model/src/cargo_workspace.rs

```diff
@@ -74,6 +75,7 @@ impl TargetKind {
                 "test" => TargetKind::Test,
                 "bench" => TargetKind::Bench,
                 "example" => TargetKind::Example,
+                "proc-macro" => TargetKind::ProcMacro,
```

Comment:
```suggestion
                "proc-macro" => TargetKind::Lib,
```

There's some confusion here as to what `Kind` means. THere are two mostly orthogonal  kinds:

* the argument you need to pass to Cargo (--lib, --bin, --example)
* --crate-type argument to Rustc.

TargetKind is for the former use-case. Might be a good idea to add a docsting to it